sklearn/preprocessing/data.py

```diff
@@ -2579,6 +2579,155 @@ def quantile_transform(X, axis=0, n_quantiles=1000,
                          " axis={}".format(axis))
 
 
+class BoxCoxTransformer(BaseEstimator, TransformerMixin):
+    """Apply the Box-Cox transform featurewise to make data more Gaussian-like.
+
+    The Box-Cox transformation is a monotonic transformation that is
+    applied to make data more Gaussian-like. This is useful for solving
+    modeling issues related to heteroscedasticity (non-constant variance),
+    or other situations where normality is desired.
+
+    Box-Cox requires input data to be strictly positive. The optimal value
+    for the parameter, lambda, is estimated through maximum likelihood.
+
+    The Box-Cox transformation is given by::
```

Comment:
This could also remain in the user guide where we usually keep mathematical details.

You need a "Read more in the User Guide" link as in other class docstrings.